src/e3/anod/qualifiers_manager.py

```diff
@@ -0,0 +1,844 @@
+
+            # Init component_names
+            self.component_names = {}
+
+            # Build component_names
+            for component, qualifier_set in self.raw_component_decls.items():
+
+                # Make sure all the qualifier in qualifier_set has been declare
+                for qualifier_name in qualifier_set:
+                    if qualifier_name not in self.qualifier_decls:
+                        raise AnodError(
+                            f'The qualifier "{qualifier_name}" is used by the '
+                            f'component "{component}" but has not been declared.'
+                        )
+
+                # Make sure all the declared qualifiers are stated
+                for qualifier_name, qualifier in self.qualifier_decls.items():
+                    if qualifier["type"] == self.__KEY_VALUE_QUALIFIER:
+                        if qualifier_name not in qualifier_set:
+                            if qualifier["default"] is None:
+                                raise AnodError(
+                                    f'The qualifier "{qualifier_name}" has been '
+                                    f"declared without a default value but is not "
+                                    f'stated for component "{component}"'
+                                )
+
+                qualifier_configuration = self.__key(qualifier_set)
+
+                # Make sure the configuration has not been used yet
+                if qualifier_configuration in self.component_names:
+                    raise AnodError(
+                        f"The qualifier configuration of {qualifier_name} is already "
+                        f"used by {self.component_names[qualifier_configuration]}."
+                    )
+
+                self.component_names[qualifier_configuration] = component
+
+    def parse(self, passed_qualifiers: dict[str, str]) -> None:
+        """Parse the provided qualifiers.
+
+        This function first makes sure that all the passed_qualifiers
+        follow the rules:
+        * Have been declared.
+        * Have been stated if they don't have a default value.
+        * Have a value which respect the choices attribute.
+
+        After the first call to this method, it is no longer possible to declare new
+        component nor qualifier. This ensure the consistency of the generated names.
+
+        After the call to parse, the qualifier_values are available using
+        __getparsed_qualifiers.
+
+        :param passed_qualifiers: a dictionnary containing the passed qualifier
+            values.
+        """
+        # End the declaration phase to guaranty the consistency of the result
+        self.__end_declaration_phase()
+
+        # Check that the received passed_qualifiers are consistent with the set of
+        # declared qualifiers.
+        self.__check_qualifier_consistency(passed_qualifiers)
+
+        # passed_qualifiers is valid and now need to be populated with the default
```

Comment:
now need to be populated is also strange is that context

Add default values when required